src/components/CohortBuilder/Summary/FileBreakdown.js

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import styled from 'react-emotion';
+import { withTheme } from 'emotion-theming';
+import { compose } from 'recompose';
+
+import BaseDataTable from 'uikit/DataTable';
+
+const FileBreakdownWrapper = styled('div')`
+  margin-top: 10px;
+`;
+
+const TableFooter = styled('div')`
+  text-align: right;
+  padding-top: 13px;
+  font-size: 11px;
+  font-weight: bold;
+  color: ${({ theme }) => theme.secondary};
+  & a {
+    font-weight: normal;
+    color: ${({ theme }) => theme.hover};
+    text-decoration: underline;
+    margin-left: 5px;
+  }
+`;
+
+const Column = styled('span')`
+  font-size: 11px;
+`;
+
+const FilesColumn = styled(Column)`
+  color: ${({ theme }) => theme.hover};
+  text-decoration: underline;
+`;
+
+const sumTotalFilesInData = dataset => {
+  return dataset.reduce((accumulator, datum) => {
+    return accumulator + parseInt(datum.files);
+  }, 0);
+};
+
+const SEARCH_FILE_RELATIVE_URL = '/search/file';
+
+const generateFileRepositoryUrl = (dataType, experimentalStrategy) => {
+  return (
+    `${SEARCH_FILE_RELATIVE_URL}?sqon=` +
+    encodeURI(
+      JSON.stringify({
+        op: 'and',
+        content: [
+          {
+            op: 'in',
+            content: {
+              field: 'data_type',
+              value: [dataType],
+            },
+          },
+          {
+            op: 'in',
+            content: {
+              field: 'sequencing_experiments.experiment_strategy',
+              value: [experimentalStrategy],
+            },
+          },
+        ],
+      }),
+    )
+  );
+};
+
+const localizeFileQuantity = quantity => {
+  return `${Number(quantity).toLocaleString()} file${quantity > 1 ? 's' : ''}`;
+};
+
+const generateFileColumnContents = dataset => {
+  return dataset.map(datum => {
+    datum.fileLink = (
+      <a href={generateFileRepositoryUrl(datum.dataType, datum.experimentalStrategy)}>
+        {localizeFileQuantity(datum.files)}
+      </a>
+    );
```

Comment:
let's not mutate data in map. The following is preferred:
```
return {
    ... datum,
    fileLink: (
        <a href={generateFileRepositoryUrl(datum.dataType, datum.experimentalStrategy)}>
            {localizeFileQuantity(datum.files)}
        </a>
    )
}
```

Reply:
Done'd. @hlminh2000 Please approve.